numalogic/registry/redis_registry.py

```diff
@@ -196,16 +202,19 @@ def __save_artifact(
         latest_key = self.__construct_latest_key(key)
         pipe.set(name=latest_key, value=new_version_key)
         _LOGGER.debug("Setting latest key : %s ,to this new key = %s", latest_key, new_version_key)
-        serialized_metadata = ""
-        if metadata:
-            serialized_metadata = orjson.dumps(metadata)
+        serialized_metadata = orjson.dumps(metadata) if metadata else ""
         serialized_artifact = dumps(deserialized_object=artifact)
+        _cur_ts = int(time.time())
         pipe.hset(
             name=new_version_key,
             mapping={
                 "artifact": serialized_artifact,
-                "version": str(version),
-                "timestamp": time.time(),
+                "version": version,
+                "timestamp": random.randrange(
+                    _cur_ts - self.jitter_secs,
+                    _cur_ts + self.jitter_secs + 1,
+                    60 * self.jitter_steps,
```

Comment:
optionally, we can have a helper function such as ```value = apply_jitter(value, jitter, step)```.

related with my earlier comments, if jitter_steps is truly in minute unit, suggest to name the variable as ```jitter_step_minutes``` which makes the difference with ```jitter_secs``` immediately clear.

Reply:
Yes. It should be in minutes . That is a fair ask. Let me change the variable name to have minute info.